scripts/assemble-proto.gradle

```diff
@@ -0,0 +1,140 @@
+import com.google.common.base.Predicate
+import com.google.common.collect.Iterables
+import com.google.common.io.Files
+
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+buildscript {
+    repositories {
+        mavenCentral()
+        mavenLocal()
+    }
+
+    dependencies {
+        classpath "com.google.guava:guava:$guavaVersion"
+    }
+}
+
+/**
+ * Collects all the directories from current project and its dependencies (including zip tree
+ * directories) which contain {@code .proto} definitions.
+ *
+ * <p>The directories may in practice include files of other extension. The caller should take care
+ * of handling those files respectively.
+ *
+ * <p>It's guaranteed that there are no other Proto definitions in the current project classpath
+ * except those included into the returned {@code Collection}.
+ */
+Collection<File> collectProto() {
+    final def dependencies = configurations.compile.files
+    final def jarFiles = dependencies.collect { it.name }
+    final def result = new HashSet<>()
+    for (final File jarFile in dependencies) {
+        if (jarFile.name.endsWith(".jar")) {
+            final def zipTree = zipTree(jarFile)
+            for (final File file in zipTree) {
+                if (IsProtoFile.PREDICATE.apply(file)) {
+                    result.add(getProtoRoot(file, jarFiles))
+                }
+            }
+        }
+    }
+    result.addAll(sourceSets.main.proto.srcDirs)
+    return result
+}
+
+/**
+ * Returns the root directory containing a Proto package.
+ *
+ * @param member the member File of the Proto package
+ * @param jars the full listing of the project JAR dependencies
```

Comment:
Having those as strings is fragile. At least describe and check the format. It's of course better to have a value object instead of a `String`.

Also, I wouldn't call it jars. They are jarPaths or something like that (depending on the value object name).